src/parser.rs

```diff
@@ -708,7 +709,7 @@ impl Parser {
                     Ok(20)
                 }
                 Token::Plus | Token::Minus => Ok(Self::PLUS_MINUS_PREC),
-                Token::Mult | Token::Div | Token::Mod => Ok(40),
+                Token::Mult | Token::Div | Token::Mod | Token::StringConcat => Ok(40),
```

Comment:
Would you comment why you picked this precedence, even if it's because we don't care about expressions combining `||` with other operations.

Reply:
BigQuery has a list of operator precedence for "their" ansi sql implementation.

https://cloud.google.com/bigquery/docs/reference/standard-sql/operators

Reply:
Even though, I agree that you don't need it when you wil not combine it with other string operators.